src/Parser.php

```diff
@@ -115,6 +113,8 @@ protected function buildFromString(string $grammar, array $options)
      */
     public function __construct($grammar, array $options = [])
     {
+        $options += $this->getDefaultOptions();
```

Comment:
The array union construct does not overwrites keys already present in `$options`; quick way to apply defaults in PHP

Reply:
Great change but it must be changed a bit.
Name $options is probably not proper as it's not only containing options.
At this point it contains 4 options and one extra data which is "parser".
$options['parser'] should always be set to $this

Reply:
> $options['parser'] should always be set to $this

Make sense! I wasn't 110% sure so I didn't change it so far.

Now: I just pushed another commit which basically does this:
- keep `$options` and `'parser'` (for now)
- make it not really "configurable" from the outside, always instantly overwrite/set in the constructor)
- adapt code so it can always expect the parser (no `if (isset($options['parser']…`)
- add TODO for future improvement
- removed it from `getDefaultOptions()` to make it clear it's not an "option"

Does that sound acceptable for now?